### PR TITLE
test against pre-built binaries of `numpy`

### PR DESCRIPTION
**Pull request by zacharyburnett**
Resolves [SCSB-69](https://jira.stsci.edu/browse/SCSB-69)

test against `pre` wheels of `astropy` and `numpy` to ensure incoming issues are resolved

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -3,5 +3,6 @@ git+https://github.com/asdf-format/asdf-standard
 git+https://github.com/asdf-format/asdf-transform-schemas
 git+https://github.com/asdf-format/asdf-coordinates-schemas
 git+https://github.com/asdf-format/asdf-wcs-schemas
-git+https://github.com/astropy/astropy
+astropy>=0.0.dev0
 git+https://github.com/astropy/asdf-astropy
+numpy>=0.0.dev0
```
Comment (WilliamJamieson) [inline requirements-dev.txt:8]: This isn't quite right, see https://github.com/spacetelescope/romancal/pull/695#discussion_r1189076034 and https://github.com/spacetelescope/romancal/pull/695#discussion_r1189078751 for details.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

**Comment by zacharyburnett** [comment]
GitHub Actions are currently degraded; running `tox -e test-devdeps-xdist` locally results in the following `pip freeze`: 
```
asdf @ git+https://github.com/asdf-format/asdf@3d46a99c69a06dcb806606190cc7815e15483a3c
asdf-astropy @ git+https://github.com/astropy/asdf-astropy@52d202781b4ecb3e17da052d82f0a8516f5b35b1
asdf-coordinates-schemas @ git+https://github.com/asdf-format/asdf-coordinates-schemas@533d6eee646c197bd2ce7d818596c861c1b692b6
asdf-standard @ git+https://github.com/asdf-format/asdf-standard@0d4e214b5101aa6a458eee57b4f756e99bb840cb
asdf-transform-schemas @ git+https://github.com/asdf-format/asdf-transform-schemas@f2f4ab88c3c88febf03aba2a97bcd273b3029406
asdf-unit-schemas==0.1.0
asdf-wcs-schemas @ git+https://github.com/asdf-format/asdf-wcs-schemas@1696d41cc4d621791ce18239bca6360ba81abf3e
astropy @ git+https://github.com/astropy/astropy@f2fbb112983e953fb772c5ac81ec50dfa34bcc19
attrs==23.1.0
certifi==2023.5.7
charset-normalizer==3.1.0
ci-watson==0.6.1
coverage==7.2.5
crds==11.17.0
execnet==1.9.0
filelock==3.12.0
gwcs @ file:///Users/zburnett/projects/gwcs/.tox/.tmp/package/44/gwcs-0.18.4.dev29%2Bg6aba5ba.tar.gz#sha256=5c86dadad33407950bd3437b58ace7bb6a672dc967c4c02439da31d390d210ce
hypothesis==6.75.2
idna==3.4
importlib-metadata==6.6.0
iniconfig==2.0.0
jmespath==1.0.1
jsonschema==4.17.3
numpy==1.25.0.dev0+1357.ga2d21d8ac
packaging==23.1
Parsley==1.3
pluggy==1.0.0
psutil==5.9.5
pyerfa==2.0.0.3
pyrsistent==0.19.3
pytest==7.3.1
pytest-arraydiff==0.5.0
pytest-astropy==0.10.0
pytest-astropy-header==0.2.2
pytest-cov==4.0.0
pytest-doctestplus==0.12.1
pytest-filter-subpackage==0.1.2
pytest-mock==3.10.0
pytest-openfiles==0.5.0
pytest-remotedata==0.4.0
pytest-xdist==3.2.1
PyYAML==6.0
requests==2.30.0
scipy==1.10.1
semantic-version==2.10.0
sortedcontainers==2.4.0
urllib3==2.0.2
zipp==3.15.0
```

**Comment by braingram** [comment]
@zacharyburnett the `test-dev` job is failing due to `asdf-astropy` requiring an update to deal with upstream changes in `astropy` (a few classes were moved requiring updates to the converters). A PR is open to fix the changes:
https://github.com/astropy/asdf-astropy/pull/181
I ran the gwcs tests locally. Without the above asdf-astropy PR I see the error seen here in `test-dev`. When run with the asdf-astropy PR no error was reported.

File: tox.ini [hunk 2]
```diff
@@ -57,14 +57,15 @@ passenv =
     CI
     CODECOV_*
     DISPLAY
+set_env =
+    dev: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/scipy-wheels-nightly/simple
```
Comment (zacharyburnett) [inline tox.ini:61]:
```suggestion
    dev: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
```

**Review by nden — APPROVED** [review]

**Pull request merged**